### ISSUE TITLE
prune optimizer failed

### ISSUE THREAD

**Issue by TheCrappyActionFigureMaster** [SJoJoK/3DGStream#36]
Hello, first of all, thank you for the great work!

i just have one problem, 
after tweaking with some parameters, i get "prune optimizer failed", and when i print out the stored_state["exp_avg"].size(0), its 0.

It almost always happens if my s1 iterations are small (<50) and sometimes happened when s1 iteration is big (>300).

Is this because i need to forward + backward + optimizer.step() before i add and prune? 

or is there something wrong with the parameters I'm not aware of?

    --densify_from_iter 130 \
    --spawn_type "spawn" \
    --s2_adding \
    --densification_interval 20 \
    --densify_grad_threshold 0.00015 \
    --min_opacity 0.005 \
    --s2_type "split" \
    --std_scale 1 \
    --num_of_split 1 \
    --num_of_spawn 2 \
    --extent 0 \

thank you very much in advance!!
keep up the good work!!

update:

i realize when that happen, the loss was super high in stage 1 ( > 0.2 ), and hence also high loss in stage 2, with prune error.
so i suppose it was something else gone wrong during training in stage 1, and happens quit randomly but started to happen when i implement update_lr in each iteration.
